package/com.unity.formats.usd/Common/UsdEditorAnalytics.cs

```diff
@@ -0,0 +1,199 @@
+# define USE_EDITOR_ANALYTICS
+# endif
+
+// Copyright 2023 Unity Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Analytics;
+
+namespace Unity.Formats.USD
+{
+    public static class UsdEditorAnalytics
+    {
+        // General analytics
+        const int k_MaxEventsPerHour = 1000;
+        const int k_MaxNumberOfElements = 100;
+        const string k_VendorKey = "unity.formats.usd";
+
+        // Universal USD Analytics
+        const string k_UsageEventName = "uUSDPackageUsage";
+
+        struct UsageAnalyticsData
+        {
+            public bool InitSucceeded;
+            public float TimeTakenS;
+        }
+
+        // USD Import Analytics
+        const string k_ImportEventName = "uUSDFileImport";
+
+        struct ImportAnalyticsData
+        {
+            public string FileExtension;
+            public float TimeTakenS;
+            public bool ImportSucceeded;
+            public bool IncludesMeshes;
+            public bool IncludesPointInstancer;
+            public bool IncludesMaterials;
+            public bool IncludesSkel;
+        }
+        public struct ImportResult
+        {
+            public bool Success;
+            public bool ContainsMeshes;
+            public bool ContainsPointInstancer;
+            public bool ContainsSkel;
+            public bool ContainsMaterials;
+
+            public static ImportResult Default => new ImportResult()
+            {
+                Success = false,
+                ContainsMeshes = false,
+                ContainsPointInstancer = false,
+                ContainsSkel = false,
+                ContainsMaterials = false
+            };
+        };
+
+        // USD Export Analytics
+        const string k_ExportEventName = "uUSDFileExport";
+
+        struct ExportAnalyticsData
+        {
+            public string FileExtension;
+            public float TimeTakenS;
+            public bool ExportSucceeded;
+            public bool OnlyOverrides;
+        }
+
+        // USD Recorder Export Analytics
+        const string k_RecorderExportEventName = "uUSDFileRecorderExport";
+
+        struct RecorderExportAnalyticsData
+        {
+            public string FileExtension;
+            public bool ExportSucceeded;
+            public bool OnlyOverrides;
+            public int FrameCount;
```

Comment:
Should we be recording time taken here too or is that not valid in this context?